### ISSUE TITLE
[BUG?] ESLint is checking syntax for text inside HTML element, when it shouldn't.

### ISSUE THREAD

**Issue by trusktr** [yannickcr/eslint-plugin-react#649]
For example, I have a file with this content:

```js
import jss from 'src/lib/jss'

let style = <style>
  #navbar-logo{
    margin-top: -4px;
  }

  .register-button:hover{
    color: white;
    -webkit-box-shadow: 1px 1px 1px 0px rgba(0,0,0,0.75);
    -moz-box-shadow: 1px 1px 1px 0px rgba(0,0,0,0.75);
    box-shadow: 1px 1px 1px 0px rgba(0,0,0,0.75);
  }

  #permission-disclamer{
    font-size: 10px;
    color: grey;
  }

  /* Custom, iPhone Retina */
  @media only screen and (min-width : 320px) {
    #permission-disclamer{
      width: 278px;
      margin: 0 auto;
      margin-top: 20px;
      text-align: center;
    }
  }

  /* MD Medium Devices, Desktops */
  @media only screen and (min-width : 992px) {
    #permission-disclamer{
      width: 278px;
      margin-left: 6px;
      margin-top: 10px;
      text-align: left;
    }
  }
</style>

jss.createStyleSheet(style, {named: false}).attach()
```

On line 5, which says `margin-top: -4px;`, ESLint says `Parsing error: Unexpected token`, but ESLint should not care what text I put inside of an HTML element (JSX).

My eslintrc:

```js
module.exports = {
    extends: "eslint:recommended",
    installedESLint: true, // ??? what's this for?
    parser: 'babel-eslint',
    parserOptions: {
        ecmaFeatures: {
            experimentalObjectRestSpread: true,
            jsx: true,
            modules: true, // import/export statements
        },
        sourceType: "module"
    },
    env: {
        browser: true,
        commonjs: true,
        es6: true,
        node: true
    },
    plugins: [
        "react"
    ],
    rules: {
        indent: [ "off", 4, {SwitchCase: 1} ],
        'linebreak-style': [ "error", "unix" ],
        //quotes: [ "error", "single" ],
        //semi: [ "error", "never" ]
        'no-console': 'off',
        'comma-dangle': 'off',
        'no-unused-vars': 'warn',
        'no-constant-condition': 'off',
    }
};
```

Did I miss some configuration perhaps?

**Comment by ljharb** [comment]
Aren't the `{` in jsx interpreted as surrounding a JS character? I think you might need `{'{'}` and `{'}'}` to have literal curly braces.

**Comment by trusktr** [comment]
@ljharb Ah,that might be why, I didn't think about that. However if I do that then I think it will break [CSSX](https://github.com/krasimir/cssx) the transpile step. cc @krasimir

**Comment by ljharb** [comment]
If so, that sounds like a fundamental incompatibility between CSSX and JSX - I don't think that's something eslint-plugin-react can really fix :-/

**Comment by krasimir** [comment]
@trusktr I just made [this](https://github.com/krasimir/eslint-plugin-cssx/commit/4035029c4c0c84351e117ed7660e74061d1b397a) commit. As a source for the new test case I used your code above and the test passes. I thought that maybe the order of running the plugins matter but that wasn't the case. No matter if I place `eslint-plugin-react` first or `eslint-plugin-cssx` the test passes without any errors. Is it possible that it's something in your setup that causes `eslint-plugin-cssx` to not be part of the process.
